lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -46,14 +46,22 @@ def call(env)
           # configure the Rack middleware once
           configure()
 
-          # start a new request span and attach it to the current Rack environment;
-          # we must ensure that the span `resource` is set later
-          request_span = @tracer.trace(
-            'rack.request',
+          trace_options = {
             service: @service,
             resource: nil,
             span_type: Datadog::Ext::HTTP::TYPE
-          )
+          }
+
+          # Merge distributed trace ids if present
+          unless env['HTTP_X_DDTRACE_PARENT_TRACE_ID'].nil? || env['HTTP_X_DDTRACE_PARENT_SPAN_ID'].nil?
```

Comment:
Hi all, indeed the plan is to have these headers be the same for all languages, obviously. Sorry for the delay, we're still discussing a few points here, as we know once this is officially released, there's no way back, any update would be a breaking change for you. Still, I understand the frustration of maintaining your fork @cabello ... BTW @whithajess this should probably be merged/rebased with some upcoming PR which changes the way we deal with the current span. I'm leaving a small comment in the code about this. Thanks for your patience, we'll get this one moving on, for sure!